osu.Game/Screens/Play/PauseContainer.cs

```diff
@@ -43,24 +43,32 @@ public int Retries
         public Action OnRetry;
         public Action OnQuit;
 
-        private readonly FramedClock framedClock;
-        private readonly DecoupleableInterpolatingFramedClock decoupledClock;
+        private readonly FramedClock offsetClock;
+        private readonly DecoupleableInterpolatingFramedClock adjustableClock;
+
+        /// <summary>
+        /// The final clock which is exposed to underlying components.
+        /// </summary>
+        [Cached]
+        private readonly GameplayClock gameplayClock;
 
         /// <summary>
         /// Creates a new <see cref="PauseContainer"/>.
         /// </summary>
-        /// <param name="framedClock">The gameplay clock. This is the clock that will process frames.</param>
-        /// <param name="decoupledClock">The seekable clock. This is the clock that will be paused and resumed.</param>
-        public PauseContainer(FramedClock framedClock, DecoupleableInterpolatingFramedClock decoupledClock)
+        /// <param name="offsetClock">The gameplay clock. This is the clock that will process frames. Includes user/system offsets.</param>
+        /// <param name="adjustableClock">The seekable clock. This is the clock that will be paused and resumed. Should not be processed (it is processed automatically by <see cref="offsetClock"/>).</param>
+        public PauseContainer(FramedClock offsetClock, DecoupleableInterpolatingFramedClock adjustableClock)
         {
-            this.framedClock = framedClock;
-            this.decoupledClock = decoupledClock;
+            this.offsetClock = offsetClock;
+            this.adjustableClock = adjustableClock;
+
+            gameplayClock = new GameplayClock(offsetClock);
```

Comment:
I'd like this to be moved to Player since it's pretty hidden where it is right now, and there doesn't seem to be a particular reason why it's constructed here.

Reply:
I put it here because `offsetClock.ProcessFrame` happens in `PauseContainer`. Are you okay with the caching/blocking happening one level above the processing?

Reply:
As discussed, for the time being we will rename the container. Going forward all clock logic should be moved into a dedicated container.